obs-packaging/linux-container/debian.rules

```diff
@@ -18,6 +18,10 @@ override_dh_auto_build:
 	rm -f .config
 	find $(KERNEL_CONFIGS) -name "$(KERNEL_ARCH)_kata_kvm_*" -exec cp {} .config \;
 	[ -f .config ] || (@echo "ERROR: cannot find the kernel config file for the $(KERNEL_ARCH) architecture"; exit 1)
+	# https://github.com/kata-containers/packaging/issues/394 enable reproducible builds:
+	export KBUILD_BUILD_USER=katabuilduser
+	export KBUILD_BUILD_HOST=katabuildhost
+	export KBUILD_BUILD_TIMESTAMP="$(LANG=C date -d "@${SOURCE_DATE_EPOCH:-978310861}")"
```

Comment:
Any reason why `978310861` instead of what's returned returned by `date` at build time?

Reply:
Should be good to add a document that explain why is that number needed

Reply:
'date' at build time varies between builds, so it would not be possible to get the same binary in another build.

Instead of 978310861 we could chose any other constant - e.g. "1". The meaning becomes more obvious in the decoded form:
```bash
date -u -d @978310861
Mon Jan  1 01:01:01 UTC 2001
```

We could opt for `date -u -r $file` instead to use the mtime of a file contained in the tarball.

Reply:
There is a certain elegance with defaulting for zero, along with an explanatory comment:

```sh
# Default to zero seconds after midnight 1970-01-01 (the epoch)
export KBUILD_BUILD_TIMESTAMP="$(LANG=C date -d "@${SOURCE_DATE_EPOCH:-0}")"
 ```

But I think we also need to force the timezone to make this reproducible:

```sh
$ LANG=C date -d "@0"
Thu Jan  1 01:00:00 BST 1970    # XXX: wrong
$ LANG=C TZ=UTC date -d "@0"
Thu Jan  1 00:00:00 UTC 1970    # XXX: correct
```

Hence,

```sh
export KBUILD_BUILD_TIMESTAMP="$(LANG=C TZ=UTC date -d "@${SOURCE_DATE_EPOCH:-0}")"
```

Reply:
Using date's -u switch for UTC now.